x-pack/plugins/transform/public/app/sections/create_transform/components/step_details/step_details_form.tsx

```diff
@@ -840,6 +855,58 @@ export const StepDetailsForm: FC<StepDetailsFormProps> = React.memo(
                 data-test-subj="transformMaxPageSearchSizeInput"
               />
             </EuiFormRow>
+            <EuiFormRow
+              label={i18n.translate(
+                'xpack.transform.stepDetailsForm.transformNumFailureRetriesLabel',
+                {
+                  defaultMessage: 'Number of failure retries',
+                }
+              )}
+              isInvalid={!isTransformSettingsNumFailureRetriesValid}
+              error={
+                !isTransformSettingsNumFailureRetriesValid && [
+                  i18n.translate('xpack.transform.stepDetailsForm.NumFailureRetriesError', {
+                    defaultMessage:
+                      'Number of retries needs to be a number greater than or equal to -1. The minimum value is `0` and the maximum is `100`.\n' +
+                      '`-1` can be used to denote infinity. In this case, the transform never gives up on retrying a recoverable failure.',
+                  }),
+                ]
+              }
+              helpText={i18n.translate(
+                'xpack.transform.stepDetailsForm.transformNumRetriesHelpText',
+                {
+                  defaultMessage:
+                    'Defines the number of retries on a recoverable failure before the transform task is marked as `failed`. If not set, the cluster setting `xpack.transform.num_transform_failure_retries` or a default of 10 will be used.',
+                }
+              )}
+            >
+              <EuiFieldText
+                value={
+                  transformSettingsNumFailureRetries ||
```

Comment:
The value isn't being retained on clone. If set, the value should copied over when cloning.